MediaBrowser.Api/UserService.cs

```diff
@@ -501,27 +503,21 @@ public async Task Post(UpdateUser request)
             }
         }
 
-        /// <summary>
-        /// Posts the specified request.
-        /// </summary>
-        /// <param name="request">The request.</param>
-        /// <returns>System.Object.</returns>
         public async Task<object> Post(CreateUserByName request)
         {
-            var dtoUser = request;
+            var newUser = await _userManager.CreateUser(request.Name).ConfigureAwait(false);
 
-            var newUser = await _userManager.CreateUser(dtoUser.Name).ConfigureAwait(false);
+            // no need to authenticate password for new user
+            if (request.Password != null)
+            {
+                await _userManager.ChangePassword(newUser, request.Password).ConfigureAwait(false);
```

Comment:
Does the CreateUser call auto fail for existing users? Otherwise this might be an avenue for messing with other users' passwords.

Reply:
Just checked and the server will actually throw an exception so we should be fine. Ideally we could pass an optional password field with CreateUser to avoid accidentally causing that issue, but it would probably just call the same method internally.